torch/nn/modules/_functions.py

```diff
@@ -45,7 +46,7 @@ def forward(self, input, weight, bias, running_mean, running_var, eps, momentum,
             count_all.view(-1)
         )
 
-        self.save_for_backward(input, weight, mean, invstd, count_all)
+        self.save_for_backward(input, weight, mean, invstd, count_all.to(torch.int32))
```

Comment:
I think there were recent changes to make `count` floating point to call a single allgather? Please check and update accordingly

Reply:
Hmm, I didn't see a relevant change. If you mean this https://github.com/pytorch/pytorch/pull/50126/files, then it should still be fine.